source/components/temp_doeclim_component.cpp

```diff
@@ -0,0 +1,528 @@
+    if( tgav_constrain.size() ) {
+        Logger& glog = Logger::getGlobalLogger();
+        H_LOG( glog, Logger::WARNING ) << "Temperature will be overwritten by user-supplied values!" << std::endl;
+    }
+    
+    dt = 1;         // years per timestep (this is hard-coded into Hector
+    ak = 0.31;      // slope in climate feedback - land-sea heat exchange linear relationship
+    bk = 1.59;      // offset in climate feedback - land-sea heat exchange linear relationship, W/m2/K
+    csw = 0.13;     // specific heat capacity of seawater W*yr/m3/K
+    earth_area = 5100656 * pow(10.0, 8);
+    kcon = 3155.0;  // conversion from cm2/s to m2/yr
+    q2co = 3.7;     // radiative forcing for atmospheric CO2 doubling, W/m2
+    rlam = 1.43;    // factor between land clim. sens. and sea surface clim. sens. T_L2x = rlam*T_S2x
+    secs_per_Year = 31556926.0;
+    zbot = 4000.0;  // bottom depth of diffusive ocean, m
+    bsi = 1.3;                  // warming factor for marine surface air over SST (due to retreating sea ice)
+    cal = 0.52;                 // heat capacity of land-troposphere system, W*yr/m2/K
+    cas = 7.80;                 // heat capacity of mixed layer-troposphere system, W*yr/m2/K
+    flnd = 0.29;                // fractional land area
+    fso = 0.95;                 // ocean fractional area below 60m
+    
+    
+    // Initializing all model components that depend on the number of timesteps (ns)
+    ns = core->getEndDate() - core->getStartDate() + 1;
+    
+    KT0.resize(ns);
+    KTA1.resize(ns);
+    KTB1.resize(ns);
+    KTA2.resize(ns);
+    KTB2.resize(ns);
+    KTA3.resize(ns);
+    KTB3.resize(ns);
+    
+    Ker.resize(ns);
+    
+    temp.resize(ns);
+    temp_landair.resize(ns);
+    temp_sst.resize(ns);
+    heatflux_mixed.resize(ns);
+    heatflux_interior.resize(ns);
+    heat_mixed.resize(ns);
+    heat_interior.resize(ns);
+    forcing.resize(ns);
+    
+    for(int i=0; i<3; i++) {
+        B[i] = 0.0;
+        C[i] = 0.0;
+    }
+    
+    for(int i=0; i<ns; i++) {
+        KT0[i] = 0.0;
+        KTA1[i] = 0.0;
+        KTB1[i] = 0.0;
+        KTA2[i] = 0.0;
+        KTB2[i] = 0.0;
+        KTA3[i] = 0.0;
+        KTB3[i] = 0.0;
+    }
+    
+    // Dependent Model Parameters
+    ocean_area = (1.0 - flnd) * earth_area;
+    cnum = rlam * flnd + bsi * (1.0 - flnd);
+    cden = rlam * flnd - ak * (rlam - bsi);
+    cfl = flnd * cnum / cden * q2co / S - bk * (rlam - bsi) / cden;
+    cfs = (rlam * flnd - ak / (1.0 - flnd) * (rlam - bsi)) * cnum / cden * q2co / S + rlam * flnd / (1.0 - flnd) * bk * (rlam - bsi) / cden;
+    kls = bk * rlam * flnd / cden - ak * flnd * cnum / cden * q2co / S;
+    keff = kcon * diff;
+    taubot = pow(zbot,2) / keff;
+    powtoheat = ocean_area * secs_per_Year / pow(10.0,22);
+    taucfs = cas / cfs;
+    taucfl = cal / cfl;
+    taudif = pow(cas,2) / pow(csw,2) * M_PI / keff;
+    tauksl  = (1.0 - flnd) * cas / kls;
+    taukls  = flnd * cal / kls;
+    
+    // First order
```

Comment:
Some more documentation needed here--what's going on?

Reply:
@bvegawe can you add a few comments here? thank you!

Reply:
My guess is that this has all been explained in a paper somewhere.  We should include a pointer to the paper and document any mappings between names used in the paper and names used here that aren't obvious.